jcache/commons/src/main/java/org/infinispan/jcache/AbstractJCacheNotifier.java

```diff
@@ -30,10 +31,10 @@
  * @author Galder Zamarreño
  * @since 5.3
  */
-public class JCacheNotifier<K, V> {
+public abstract class AbstractJCacheNotifier<K, V> {
```

Comment:
I'm not sure I understand why you need to create an abstract class for this and also one for `JCacheListenerAdapter`. The two subclasses of AbstractJCacheNotifier are pretty much identical. The real difference comes in `JCacheListenerAdapter`. Also, consider whether you could use delegation instead of inheritance to solve this? IOW, you could just pass the adapter you need as parameter on construction to JCacheNotifier.